tests/net/ieee802154/l2/src/ieee802154_shell_test.c

```diff
@@ -62,6 +64,8 @@ static void scan_result_cb(struct net_mgmt_event_callback *cb, uint32_t mgmt_eve
 	zassert_mem_equal(scan_ctx->addr, expected_coordinator_address, IEEE802154_EXT_ADDR_LENGTH);
 	zassert_equal(scan_ctx->lqi, EXPECTED_COORDINATOR_LQI,
 		      "Scan did not receive correct link quality indicator.");
+	zassert_equal(scan_ctx->association_allowed, expected_association_bit,
+		      "Scan did not set the association bit correctly.");
```

Comment:
```suggestion
		      "Scan did not set the association permit bit correctly.");
```

This is not optional, as the standard clearly names is as such, see section 7.3.1.4, figure 7-7.